pandas/tests/frame/test_missing.py

```diff
@@ -872,10 +873,22 @@ def test_interp_rowwise(self):
         result = df.interpolate(axis=1, method="values")
         assert_frame_equal(result, expected)
 
+        # GH 29142: test axis names
```

Comment:
rather than adding tests like this, can you create a new test function and paramterize on the axis

Reply:
I've added a new parametrized function in the latest commit

Reply:
pytest is returning with an uncaught warning in the same file from a call to df.median(). Should I add `tm.assert_produces_warning()`?

```
pandas/tests/frame/test_missing.py::TestDataFrameMissingData::test_fillna_categorical_nan
python3.7/site-packages/numpy/lib/nanfunctions.py:1115: RuntimeWarning: All-NaN slice encountered
    overwrite_input=overwrite_input)
```